interfaces/builtin/dbus_bind.go

```diff
@@ -0,0 +1,380 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2016 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package builtin
+
+import (
+	"bytes"
+	"fmt"
+	"regexp"
+
+	"github.com/snapcore/snapd/interfaces"
+	"github.com/snapcore/snapd/release"
+)
+
+var dbusBindPermanentSlotAppArmorShared = []byte(`
+# Description: Allow owning a name on DBus public bus
+
+#include <abstractions/###DBUS_BIND_ABSTRACTION###>
+
+# register on DBus
+dbus (send)
+    bus=system
+    path=/org/freedesktop/DBus
+    interface=org.freedesktop.DBus
+    member="{Request,Release}Name"
+    peer=(name=org.freedesktop.DBus, label=unconfined),
+
+dbus (send)
+    bus=system
+    path=/org/freedesktop/DBus
+    interface=org.freedesktop.DBus
+    member="GetConnectionUnix{ProcessID,User}"
+    peer=(name=org.freedesktop.DBus, label=unconfined),
+`)
+
+var dbusBindPermanentSlotAppArmorIndividual = []byte(`
+# bind to a well-known DBus name: ###DBUS_BIND_NAME###
+dbus (bind)
+    bus=###DBUS_BIND_BUS###
+    name=###DBUS_BIND_NAME###,
+
+# Allow us to talk to dbus-daemon
+dbus (receive)
+    bus=###DBUS_BIND_BUS###
+    path=###DBUS_BIND_PATH###
+    peer=(name=org.freedesktop.DBus, label=unconfined),
+dbus (send)
+    bus=###DBUS_BIND_BUS###
+    path=###DBUS_BIND_PATH###
+    interface=org.freedesktop.DBus.Properties
+    peer=(name=org.freedesktop.DBus, label=unconfined),
+`)
+
+var dbusBindPermanentSlotAppArmorIndividualClassic = []byte(`
+# allow unconfined clients talk to ###DBUS_BIND_NAME### on classic
+dbus (receive)
+    bus=###DBUS_BIND_BUS###
+    path=###DBUS_BIND_PATH###
+    peer=(label=unconfined),
+`)
+
+var dbusBindPermanentSlotSecComp = []byte(`
+# Description: Allow owning a name on DBus public bus
+
+getsockname
+recvmsg
+sendmsg
+sendto
+`)
+
+var dbusBindConnectedSlotAppArmorIndividual = []byte(`
+# Description: Allow DBus consumer to connect to ###DBUS_BIND_NAME###
+
+# Communicate with the well-known named DBus service
```

Comment:
This seems a bit strange, or perhaps I just misunderstand what it is. The name "bind" implies we're allowing the snap to _bind_ into the given location, of course. But then what is a "consumer of a dbus bind service", and what are we allowing the plug to do here?

Note that just like we don't have a generic "file" interface, we also probably don't want a generic "dbus" interface that allows anything at all to communicate with anything at all without a slightly more well defined context.